File: chores-frontend/src/app/model/dto/task.ts
import {BaseDto} from "./base-dto";
import {Item} from "./item";
import {OccurrenceInfo} from "../occurrence";
import {ReminderInfo} from "../reminder";

export interface Task extends BaseDto {
  description?: string;

  occurrence?: OccurrenceInfo;
  reminder?: ReminderInfo;

  item?: Item;
}

File: chores-backend/src/main/java/rs/laxsrbija/chores/domain/Task.java
package rs.laxsrbija.chores.domain;

import java.time.LocalDate;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import rs.laxsrbija.chores.domain.recurrence.Recurrence;

@Data
@EqualsAndHashCode(callSuper = true)
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
public class Task extends BaseDto {

  private String description;

  private LocalDate dateCreated;

  private Item item;

  private Recurrence recurrence;

  private OccurrenceInfo occurrence;

  private List<CompletionHistoryItem> history;

  private ReminderInfo reminder;

  private boolean enabled;
}
